### PR TITLE
[DNM]Add geo-partition doc

### PR DESCRIPTION
**Pull request by djshow832**
### What is changed, added or deleted? (Required)

Add a document on Geo-Partition. Geo-Partition is only available in TiDB 5.0.
Currently, this PR is only used for testing. DO NOT MERGE.

### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [ ] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Have version specific changes 
- [ ] Might cause conflicts

### CODE REVIEW

File: configure-geo-partition.md [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+aliases: ['/docs-cn/dev/configure-geo-partition/']
+---
+
+# Geo-Partition 使用文档
+
+## 场景描述
+
+在跨地域部署的 TiDB 集群中，由于 TiDB 实例与 TiKV 实例分布在不同的地区，TiDB 常常需要跨地区访问数据，导致较高的延时。
+
+为了减少跨地区数据访问，可以人为地控制数据的位置，使得业务端离数据更近。为此，TiDB 5.0 提供了 Geo-Partition。用户把表进行分区，然后把分区和 TiKV 实例关联起来，以指导 PD 把该分区的数据调度到对应的 TiKV 实例上。
```
Comment (zz-jason) [inline configure-geo-partition.md:13]:
```suggestion
为了减少跨地区数据访问，可以人为地控制数据的位置，使得数据离业务端更近。为此，TiDB 5.0 提供了 Geo-Partition。用户把表进行分区，然后把分区和 TiKV 实例关联起来，以指导 PD 把该分区的数据调度到对应的 TiKV 实例上。
```

因为关注点是 “数据”，调整了下语句描述

File: configure-geo-partition.md [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+
+当场景同时满足以下条件时，推荐使用 Geo-Partition：
+
+- 对读或写延时要求高
+- 表内的数据与地域强关联
+- 表中有与地区相关的字段，通过该字段的值能计算出数据的放置位置
```
Comment (zz-jason) [inline configure-geo-partition.md:19]:
```suggestion
- 表中有与地区相关的字段，通过该字段的值能指导数据放置的地理位置
```

File: configure-geo-partition.md [hunk 3]
```diff
@@ -0,0 +1,194 @@
...
+
+## 运作原理
```
Comment (zz-jason) [inline configure-geo-partition.md:21]:
```suggestion
## 原理介绍
```

File: configure-geo-partition.md [hunk 4]
```diff
@@ -0,0 +1,194 @@
...
+
+TiDB 4.0 提供了 Placement Rules 功能，用户可以通过 Placement Rules 配置数据的副本数量、存放位置、主机类型等放置策略。
```
Comment (zz-jason) [inline configure-geo-partition.md:23]:
```suggestion
TiDB 4.0 中，PD 组件内部提供了 [Placement Rules](https://docs.pingcap.com/zh/tidb/stable/configure-placement-rules#placement-rules) 的功能，使得用户可以通过 Placement Rules 配置数据的副本数量、存放位置、主机类型等放置策略。
```

File: configure-geo-partition.md [hunk 5]
```diff
@@ -0,0 +1,194 @@
...
+
+TiDB 5.0 的 Geo-Partition 是在 Placement Rules 的基础上，提供了 SQL 接口，方便用户新增、更改、删除、查看分区的放置策略。用户在定义、更改分区的放置策略，或是执行分区相关的 DDL 语句时，TiDB 实例生成对应的 Placement Rules，并向 PD 发送更新请求。
```
Comment (zz-jason) [inline configure-geo-partition.md:25]:
```suggestion
TiDB 5.0 的 Geo-Partition 是在 Placement Rules 的基础上，提供了 SQL 语法，方便用户新增、更改、删除、查看分区的放置策略。用户在定义、更改分区的放置策略，或是执行分区相关的 DDL 语句时，TiDB 实例生成对应的 Placement Rules，并向 PD 发送更新 Placement Rules 的请求。PD 根据用户在 SQL 语句配置的 Placement Rule 执行相应的数据放置策略。
```

File: configure-geo-partition.md [hunk 6]
```diff
@@ -0,0 +1,194 @@
...
+
+- 新增副本放置策略
+- 更改副本放置策略
+- 删除副本放置策略
+
+这 3 种操作都只能通过 `ALTER TABLE table_name ALTER PARTITION partition_name` 语句实现，只是后面的子句不同。
+
+本节先介绍以上 3 种操作的语法，再依次介绍子句中每个选项的使用方法。
+
+### 新增副本放置策略
```
Comment (zz-jason) [inline configure-geo-partition.md:39]: 建议和 [优化实例](https://docs.pingcap.com/zh/tidb/stable/explain-overview#%E4%BC%98%E5%8C%96%E5%AE%9E%E4%BE%8B) 一样，准备一个实际例子，手把手教用户如何：

1. 部署集群，配置 label
2. 建表，导数据
3. 配置副本放置策略
4. 对比配置副本放置策略前后业务上的优化效果

File: configure-geo-partition.md [hunk 7]
```diff
@@ -0,0 +1,194 @@
...
+	...
+```
+
+该语句会在原来的副本数量基础上，再给 `partition_name` 增加 `replicas` 个 `role` 角色的副本，包括索引。当存在 `CONSTRAINT` 选项时，新增的副本要满足 `CONSTRAINT` 里的限制。
+
+可以在一条 SQL 语句里定义多个放置策略，用逗号分隔。例如：
+
+```sql
+ALTER TABLE user ALTER PARTITION p0
+	ADD PLACEMENT POLICY CONSTRAINTS="[+zone=sh]" ROLE=follower REPLICAS=1
```
Comment (crazycs520) [inline configure-geo-partition.md:55]: 结尾少了一个 , ?

File: configure-geo-partition.md [hunk 8]
```diff
@@ -0,0 +1,194 @@
...
+	PARTITION jpn VALUES LESS THAN(4),
+	PARTITION can VALUES LESS THAN(5)
+);
+```
+
+最后把分区 `chn` 和 `jpn` 的 leader 副本放置在中国数据中心，把 `usa` 和 `can` 的 leader 放置在美国数据中心：
+
+```sql
+ALTER TABLE user ALTER PARTITION chn
+	ALTER PLACEMENT POLICY ROLE=leader CONSTRAINTS="['+zone=chn']";
```
Comment (crazycs520) [inline configure-geo-partition.md:187]: 我本地这个 SQL 执行报错

```sql
> ALTER TABLE user ALTER PARTITION chn ALTER PLACEMENT POLICY ROLE=leader CONSTRAINTS="['+zone=chn']";
(8234, "Invalid placement policy 'alter placement policy constraints='[''+zone=chn'']' role=leader replicas=0': array CONSTRAINTS should be with a positive REPLICAS")

```

**Comment by qiancai** [comment]
2021-03-11: According to the PR author @lucklove, the doc changes of this PR will not be part of 5.0 GA, and there is no clear plan for these changes.

**Comment by qiancai** [comment]
2021-05-31: According to the PR author @djshow832, the doc changes of this PR will not be part of 5.1, and there is no clear plan for these changes.

**Comment by djshow832** [comment]
Deprecated.

**Pull request closed without merging**